File: Programacao-III/aula3/Forma.java
package aula3;

public abstract class Forma implements Comparable<Forma>{
    protected Ponto center;

    public Ponto getCenter() { return this.center; }
    public void setCenter(Ponto center) { this.center = center; }

    public abstract double area();
    
    public abstract double perimeter();

    public int compareTo(Forma f) {
        if(f == null)
            throw new NullPointerException("A Forma é nula");
        return (int)(area() - f.area());
    }
}

File: LFA/Projeto/geometrics-lfa-07-master/pygame/Forma.py
import abc
from Ponto import *
class Forma(metaclass = abc.ABCMeta):
    @abc.abstractmethod

    def __setCenter__(self,center):
        pass

    def __getCenter__ (self):
        pass

    #abstract method for area
    def area(self):
        pass
    
    #abastract method for perimeter
    def perimeter(self):
        pass

    def __intercept__(self,other):
        pass

    def __getColor__(self):
        pass

    def __draw__(self,surface):
        pass